pisa/core/param.py

```diff
@@ -862,6 +862,33 @@ def __init__(self, *args):
         # I think because the changed units are cached in the object (Philipp)
         self.normalize_values = True
 
+        # store list of daemonflux params names for prior penalty calculation
+        self._daemon_names = ['K_158G',
```

Comment:
To ensure future compatibility it would be good read the daemonflux parameter names from `Flux().params.known_parameters` instead of a hard coded list. If, for some reason, you want to keep a fixed parameter list, you may want to implement a check if what is returned by `Flux().params.known_parameters` is the same as what you expect it to be.

Reply:
to replace the names use simple string ops:

```python
self._daemon_names = [
     p.replace("pi+","pi").replace("pi-","antipi").replace(
          "K+","K").replace("K-","antiK")
      for p in Flux().params.known_parameters
]```

Reply:
The idea was to limit `daemonflux` dependency to the corresponding stage, that's why I wanted to avoid calling it directly in ParamSet class (that's also why chi2 value is stored into a parameter inside the stage). But I also had to add `daemonflux` to required packages anyway for automatic installation to work and now to restrict the version, so maybe there is no harm in doing it this way.
I will look at this tomorrow.

I didn't think about renaming the parameters automatically, that's a good suggestion, thank you!

Reply:
Maybe another solution which avoids storing the parameter list altogether is to just add a prefix to all daemonflux parameters names, since we have to rename them anyway to remove symbols, then I can just look for that prefix as a condition to skip these parameters...

Reply:
maybe I should just do that

Reply:
Yeah, not having a hard coded list in param.py would be better